Assets/_Scripts/Enemy/EnemyManager.cs

```diff
@@ -118,8 +121,10 @@ public void SpawnEnemies()
             }
 
             // Spawn the chosen enemy at the chosen location
-            Instantiate(chosenInfo.Prefab, spawnLocation.position, Quaternion.identity);
-
+            Enemy enemy = Instantiate(chosenInfo.Prefab, spawnLocation.position, Quaternion.identity);
+            
+            // Attaches handler to EnemyKilled event
+            enemy.EnemyKilled += _onEnemyKilled;

```

Comment:
I'm still learning handlers and listeners, so I would like to learn more about this.  Because the scripts aren't linked, do we need a listener to listen out for the event that's being triggered so it can tell the handler to handle that event? or is just having the handler enough for this situation?

Reply:
The listener I think you're mentioning here (like in the [CanvasManager](https://github.com/SolomanDavis/VGP145-S24-capstone-retro/blob/dfd1864201605e9da0d9570a0a28e1f79a1601b4/Assets/_Scripts/Managers/CanvasManager.cs#L56)) is a different way of implementing the same underlying concepts.

In the `CanvasManager`, the Unity `Button` classes have `.onClick` `UnityEvent`s that are triggered when the button is pressed [[docs](https://docs.unity3d.com/2018.3/Documentation/ScriptReference/UI.Button-onClick.html)]. The `UnityEvent` class exposes a `.addListener` function that can be used to register event handlers with the `UnityEvent` and will be called when the `UnityEvent` is triggered.

These `UnityEvent`s are provided by Unity and, if I understand right, are heavier to use than what we have here for `onEnemyKilled`.

The `EnemyKilled` event being added [in this PR](https://github.com/SolomanDavis/VGP145-S24-capstone-retro/pull/34/files#diff-51fa0e1bd3fa49668a9851b748530874a7f033df9b5895aeb4c359aa0d35e53fR21) is a native C# event of type `UnityAction`. `UnityAction` is a class provided by unity that is a zero-argument delegate (a C# concept that essentially means a way to distribute calls to other functions). These are used under the hood by `UnityEvent`s. What we are doing here though is avoiding the use of `UnityEvent`s due to their heavier nature and using the simpler and more native experience provided by C#'s `event` keyword.

So the `Enemy` class exposes a C# event called `EnemyKilled` and raises it when it itself is killed. Other classes that see the `Enemy`, such as the `EnemyManager` here, can register (via `+=` which is essentially doing the same job as `.addListener()`) event handlers such as `onEnemyKilled` with the `EnemyKilled` event. In this scenario, the `Enemy` is the "Publisher/Producer" of the `EnemyKilled` event and the `EnemyManager` is the "Subscriber/Consumer" of the event.

hopefully this answers your question 👍